File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/detail/NetsOnePresenter.java
package xuk.imaginary.gui.nets.detail;

import xuk.imaginary.data.source.NetsRepository;

import io.reactivex.android.schedulers.AndroidSchedulers;

import static xuk.imaginary.common.Utils.checkNotNull;

/**
 * @author xuie
 * @date 17-7-5
 */
public class NetsOnePresenter implements NetsOneContract.Presenter {
    private NetsRepository netsRepository;
    private NetsOneContract.View newsOneView;
    private int currentPage = 1;

    public NetsOnePresenter(NetsRepository netsRepository, NetsOneContract.View newsOneView) {
        this.netsRepository = checkNotNull(netsRepository);
        this.newsOneView = checkNotNull(newsOneView);
        this.newsOneView.setPresenter(this);
    }

    @Override
    public void subscribe() {
    }

    @Override
    public void unsubscribe() {

    }

    @Override
    public void getNewsOneRequest(String postId) {
        netsRepository.getNewDetail(postId)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(netsDetail -> newsOneView.refreshNewsOne(netsDetail),
                        Throwable::printStackTrace);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/detail/NetsOnePresenter.kt
package xuk.imaginary.gui.nets.detail

import io.reactivex.android.schedulers.AndroidSchedulers
import xuk.imaginary.data.source.NETSRepository

/**
 * Created by xuie on 17-7-5.
 */

class NetsOnePresenter(
    private val netsRepository: NETSRepository,
    private val newsOneView: NetsOneContract.View
) : NetsOneContract.Presenter {
  private val currentPage = 1

  init {
    this.newsOneView.presenter = this
  }

  override fun start() {
  }

  override fun getNewsOneRequest(postId: String) {
    netsRepository.getNewDetail(postId)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe({ netsDetail -> newsOneView.refreshNewsOne(netsDetail) },
            { e -> e.printStackTrace() })
  }
}
